src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -188,6 +188,20 @@ bool ParserIdentifier::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
         ++pos;
         return true;
     }
+    else if (pos->type == TokenType::StringLiteral)
+    {
+        ReadBufferFromMemory buf(pos->begin, pos->size());
+        String s;
+
+        readQuotedStringWithSQLStyle(s, buf);
```

Comment:
This is a low-level function, but we also have ParserStringLiteral; maybe it will fit better.